app/src/main/java/com/alexbar/layoutbasic/MainActivity.kt

```diff
@@ -19,20 +19,30 @@ import androidx.compose.material3.Scaffold
 import androidx.navigation.compose.rememberNavController
 import com.alexbar.layoutbasic.movies_app.nav.BottomNavigationBar
 import com.alexbar.layoutbasic.movies_app.nav.BottomNavigationGraph
+import com.alexbar.layoutbasic.movies_app.nav.Screens
+import com.alexbar.layoutbasic.movies_app.utils.NotificationUtils
 
 class MainActivity : ComponentActivity() {
     @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        NotificationUtils.showNotification(this)
+
         setContent {
             val navController = rememberNavController()
-
-            Scaffold(
-                bottomBar = {
-                    BottomNavigationBar(navController = navController)
+            val isLoggedIn = true // This value should be obtained from local storage
+
+            if (isLoggedIn) {
+                Scaffold(
+                    bottomBar = {
+                        BottomNavigationBar(navController = navController)
+                    }
+                ) {
+                    BottomNavigationGraph(navController)
                 }
-            ) {
+            } else {
                 BottomNavigationGraph(navController)
+                navController.navigate(Screens.LoginScreen.route) { }
             }
```

Comment:
@ezatpanah I wanted to add conditional navigation to my app in such a way that if the user is already logged in, they would be directed to the main screen; otherwise, they would be taken to the login screen. Would this be the right place to validate that part? Obviously, I would need to implement the logic to save/retrieve a value indicating that the user has already logged in. For now, I was simply hardcoding the value to test that the app navigates to the correct screen

Reply:
the place of your code is correct I have one point, in your code make sure that handles navigation properly. In your current code, both branches of the if statement seem to set up navigation, which could lead to unexpected behavior.
here is my suggestion , please check it :

```
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        NotificationUtils.showNotification(this)

        setContent {
            val navController = rememberNavController()
            val isLoggedIn = checkIfUserIsLoggedIn() // You should implement this logic to check if the user is logged in

            NavHost(navController, startDestination = if (isLoggedIn) Screens.MainScreen.route else Screens.LoginScreen.route) {
                composable(Screens.MainScreen.route) {
                    MainScreen(navController = navController)
                }
                composable(Screens.LoginScreen.route) {
                    LoginScreen(navController = navController)
                }
            }
        }
    }
}

```
and

```
sealed class Screens(val route: String) {
    object MainScreen : Screens("main_screen")
    object LoginScreen : Screens("login_screen")
}
```

By separating concerns and ensuring proper navigation logic, your code becomes easier to understand, maintain, and test. :)